gunicorn_config.py

```diff
@@ -12,3 +12,47 @@
 statsd_host = "{}:8125".format(os.getenv("STATSD_HOST"))
 keepalive = 90
 timeout = int(os.getenv("HTTP_SERVE_TIMEOUT_SECONDS", 30))  # though has little effect with eventlet worker_class
+
+debug_post_threshold = os.getenv("NOTIFY_GUNICORN_DEBUG_POST_REQUEST_LOG_THRESHOLD_SECONDS", None)
+if debug_post_threshold:
+    debug_post_threshold_float = float(debug_post_threshold)
+
+    def pre_request(worker, req):
+        # using os.times() to avoid additional imports before eventlet monkeypatching
+        req._pre_request_elapsed = os.times().elapsed
+
+    def _tuples_to_lists(value):
+        if isinstance(value, tuple):
+            # convert to list for more compact (and json-able) representation
+            return list(value)
+
+        return value
+
+    def post_request(worker, req, environ, resp):
+        elapsed = os.times().elapsed - req._pre_request_elapsed
+        if elapsed > debug_post_threshold_float:
+            import json
+            import time
+
+            import psutil
+
+            # consume this iterator to give cpu_percent calculation a "start time" to work with
+            list(psutil.process_iter(["cpu_percent"]))
+            time.sleep(0.1)  # period over which to calculate cpu_percent
```

Comment:
to clarify if i've understood what is happening here - the intention here is to say "we have just had a slow request finish - now pause for 100ms and see what else is still going on on the box at the moment even after that request finished?"

which is why the example log you showed has 0% CPU for every PID?

Reply:
Not all of the logs are all-0.0.

But yes, this is effectively what it's doing.